Sources/StreamChatUI/ChatMessageList/Attachments/FileAttachmentViewInjector.swift

```diff
@@ -0,0 +1,39 @@
+//
+// Copyright © 2021 Stream.io Inc. All rights reserved.
+//
+
+import StreamChat
+import UIKit
+
+/// The delegate used `GiphyAttachmentViewInjector` to communicate user interactions.
+public protocol FileActionContentViewDelegate: ChatMessageContentViewDelegate {
+    /// Called when the user taps on attachment action
+    func handleFileAttachmentTap(_ attachment: ChatMessageFileAttachment)
+}
+
+public typealias FilesAttachmentViewInjector = _FilesAttachmentViewInjector<NoExtraData>
+
+public class _FilesAttachmentViewInjector<ExtraData: ExtraDataTypes>: _AttachmentViewInjector<ExtraData> {
+    open lazy var fileAttachmentView: _ChatMessageFileAttachmentListView<ExtraData> = {
+        let attachmentListView = _ChatMessageFileAttachmentListView<ExtraData>()
+        attachmentListView.didTapOnAttachment = { _ in
+            guard let delegate = self.contentView.delegate as? FileActionContentViewDelegate else { return }
+        }
+        return attachmentListView.withoutAutoresizingMaskConstraints
+    }()
+
+    override open func contentViewDidLayout(options: ChatMessageLayoutOptions) {
+        contentView.bubbleView?.clipsToBounds = true
+        contentView.bubbleContentContainer.insertArrangedSubview(fileAttachmentView, at: 0, respectsLayoutMargins: false)
+    }
+
+    override open func contentViewDidUpdateContent() {
+        fileAttachmentView.content = fileAttachments
+    }
+}
+
+private extension _FilesAttachmentViewInjector {
+    var fileAttachments: [ChatMessageFileAttachment] {
+        contentView.content?.fileAttachments ?? []
+    }
+}
```

Comment:
how is this shorter than:
```
fileAttachmentView.content = contentView.content?.fileAttachments ?? []
```
? :)

Reply:
It doesn't have to be the same everywhere :) In some places it works in others it doesn't.